cpp/src/arrow/compute/kernels/aggregate_basic.cc

```diff
@@ -232,9 +238,9 @@ namespace {
 
 const FunctionDoc count_doc{"Count the number of null / non-null values",
                             ("By default, non-null values are counted.\n"
-                             "This can be changed through CountOptions."),
+                             "This can be changed through ScalarAggregateOptions."),
                             {"array"},
-                            "CountOptions"};
+                            "ScalarAggregateOptions"};
 
 const FunctionDoc sum_doc{
     "Sum values of a numeric array", ("Null values are ignored."), {"array"}};
```

Comment:
Can you update the docstring here as well? (and the same for "mean" below)

Reply:
Done.